compiler/passes/insertWideReferences.cpp

```diff
@@ -959,20 +1047,20 @@ static void propagateVar(Symbol* sym) {
             //
             SymExpr* se = toSymExpr(rhs->get(1));
             debug(sym, "ref has a wide _val, src %s (%d) of addr_of must be wide\n", se->var->cname, se->var->id);
-            setWide(se);
+            setWide(def, se);
           }
           else if (isRef(sym)) {
             if (rhs->isPrimitive(PRIM_GET_MEMBER_VALUE) || 
                 rhs->isPrimitive(PRIM_GET_MEMBER)) {
               SymExpr* field = toSymExpr(rhs->get(2));
               debug(sym, "widening field ref %s (%d)\n", field->var->cname, field->var->id);
-              matchWide(sym, field->var);
+              matchWide(def, field->var);
             }
             else if (rhs->isPrimitive(PRIM_GET_SVEC_MEMBER) || 
                      rhs->isPrimitive(PRIM_GET_SVEC_MEMBER_VALUE)) {
-              widenTupleField(rhs, sym);
+              widenTupleField(rhs, def);
             }
-            else if (rhs->isResolved()) {
+            else if (rhs->isResolved() && isRef(def)) {
```

Comment:
It was accidental, and redundant with the 'isRef(sym)' check above it. I'll remove it shortly.